lib/OpenQA/Task/Git/Clone.pm

```diff
@@ -47,11 +48,42 @@ sub _git_clone_all ($job, $clones) {
 
     # iterate clones sorted by path length to ensure that a needle dir is always cloned after the corresponding casedir
     for my $path (sort { length($a) <=> length($b) || $a cmp $b } keys %$clones) {
-        my $url = $clones->{$path};
+        my $url = $clones->{$path} // '';
         die "Don't even think about putting '..' into '$path'." if $path =~ /\.\./;
+
+        my $git = OpenQA::Git->new(app => $app, dir => $path);
```

Comment:
You create a `$git` object here, and you retrieve the url.

Then you can pass that object to `_git_clone` and don't recreate it there again.
Also the url is now set in every case `_git_clone` is called, so I would remove the `if ($url)` there.